### PR TITLE
docs(readme): update readme and getting started

### PR DESCRIPTION
**Pull request by ddysher**
**What this PR does / why we need it**:

Add getting started guide to README, still missing a lot.

**Which issue(s) this PR fixes** *(optional, close the issue(s) when PR gets merged)*:

Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -126,6 +112,167 @@ func main() {
...
+It works! The above example teaches us two facts:
+
+1. Adding validation support with Nirvana is very simple
+2. 10 characters validation is not enough to prevent spam :) (checkout guide below to add your own validation)
+
+For full example code, see [validator](./examples/getting-started/validator).
+
+### Is it popular?
+
+It's time to expose some metrics to help understand and diagonse our service! Nirvana has out-of-box support for
```
Comment (kdada) [inline README.md]: `diagonse`

File: README.md [hunk 2]
```diff
@@ -126,6 +112,167 @@ func main() {
...
+INFO  0202-11:18:50.235+08 echo.go:67 | Listening on :8080
+INFO  0202-11:18:50.235+08 builder.go:163 | Definitions: 1 Middlewares: 0 Path: /echo
+INFO  0202-11:18:50.235+08 builder.go:178 |   Method: Get Consumes: [*/*] Produces: [text/plain]
+```
+
+In another terminal:
+
+```
+$ curl "http://localhost:8080/echo?msg=test"
+Key: '' Error:Field validation for '' failed on the 'gt' tag
```
Comment (kdada) [inline README.md]: The error message is unable to understand.

Comment (ddysher) [reply README.md]: filed issue: https://github.com/caicloud/nirvana/issues/126

**Pull request merged**